src/step.js

```diff
@@ -71,19 +71,32 @@ class Step {
             return Ok(ret)
         }
 
+        let circularStructureCache = []
+        const _circularStructureHandling = (_, value) => {
+            if (typeof value === 'object' && value !== null) {
+                if (circularStructureCache.includes(value)) return
+                circularStructureCache.push(value)
+            }
+            return value
+        }
+
         let ret = undefined
         this._auditTrail.description = this.description
         this._auditTrail.return = ""
         
         ret = await _runFunction()
-        if (ret !== undefined) this._auditTrail.return = JSON.parse(JSON.stringify(ret))
+        if (ret !== undefined) { 
+            this._auditTrail.return = JSON.parse(JSON.stringify(ret, _circularStructureHandling))
```

Comment:
a PR for such a complex behavior MUST be accompanied by tests

Reply:
To be more correct: any change in behavior (add or change) should be accompanied by tests

Reply:
All right. I pushed the tests for this function.